app/controllers/training_modules_users_controller.rb

```diff
@@ -2,7 +2,7 @@
 
 class TrainingModulesUsersController < ApplicationController
   respond_to :json
-  before_action :require_signed_in, only: [:create_or_update, :mark_exercise_complete]
+  before_action :require_signed_in, only: [:create_or_update, :mark_exercise_complete, :find_slide]
```

Comment:
There's no need to call `require_signed_in` for this controller action. That would limit it to logged-in users, but we don't need to keep logged out users from using it as well.

Reply:
Okay, i have made the change